src/transformers/models/rag/modeling_rag.py

```diff
@@ -703,6 +727,11 @@ def __init__(
     def set_retriever(self, retriever: RagRetriever):
         self.rag.retriever = retriever
 
+    def set_context_encoder_for_training(self,retriever_model_name):
```

Comment:
I would prefer it here to just insert the `ctx_encoder` directly. When calling a `set_...` function I would not expect the model to load a pretrained checkpoint under the hood. => Could we change it to:

```python
def set_context_encoder_for_training(self, context_encoder):
         self.rag.ctx_encoder = context_encoder
```

Reply:
Change added.